release-24.2: roachtest: refactor tests to use WorkloadNode spec

DarrylWong opened the pull request:
Backport:
  * 1/1 commits from "roachtest: refactor tests to use WorkloadNode spec" (#126506)
  * 1/1 commits from "roachtest: mark tests that use runTPCC as using workload node" (#127805)
  * 1/1 commits from "roachtest: readd erroneously deleted workload run in alterpk-bank" (#127830)
  * 1/1 commits from "roachtest/large-schema-benchmark: fix regression with the setup phase" (#127841)
  * 1/1 commits from "roachtest: fix sysbench regression when using smaller workload node" (#127984)

Please see individual PRs for details.

/cc @cockroachdb/release


Release Justification: test only change

DarrylWong committed:
roachtest: use WorkloadNode spec and cockroach workload

This change refactors roachtests to use the new WorkloadNode
cluster spec and the helpers that come with it. It also uses
smaller workload nodes for certain tests.

This change also removes unnecessary usage of the deprecated
workload, opting to use cockroach workload when possible.

DarrylWong committed:
roachtest: mark tests that use runTPCC as using workload node

runTPCC assumes that the last node is a workload node.

Fixes: #127802
Release note: none

DarrylWong committed:
roachtest: readd erroneously deleted workload run in alterpk-bank

fqazi committed:
roachtest/large-schema-benchmark: fix regression with the setup phase

Previously, the large schema benchmark was regressed because a call to
setupTPCC was incorrectly removed via #62302. To address this, this
patch restores the missing call back, which should allow this test to
function nighties again.

Fixes: #127800, #127799, #127797, #127796, #127795,
Fixes: #127794, #127793, #127792

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127968)
<!-- Reviewable:end -->

DarrylWong committed:
roachtest: fix sysbench regression when using smaller workload node

In #126506 we made tests use a default workload node of 4 CPUs to
save money. For sysbench, this was too aggressive and a few tests
saw regressions in QPS.

This change gives sysbench workload nodes 16 CPUs instead. Testing
showed that 8 CPUs is sufficient, but to play it safe, 16 CPUs
was chosen.

Fixes: none
Release note: none
Epic: none

renatolabs commented:
I'll let you merge this one first then I'll rebase #128086.

DarrylWong commented:
Thanks!
